kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/notification/NotificationManager.kt

```diff
@@ -157,7 +157,7 @@ class NotificationManager internal constructor(
     var rewindIcon: Int? = null
 
     private fun getCurrentItemHolder(): AudioItemHolder? {
-        return player.currentMediaItem?.localConfiguration?.tag as AudioItemHolder?
+        return player?.currentMediaItem?.localConfiguration?.tag as AudioItemHolder?
```

Comment:
I'm unclear as to how this fixes the problem. `player` is passed into the constructor so there should never be a time when it is `null`.

Could you help me understand what's going on here?

Reply:
Ahhh, I see now the superclass defines the `setPlayer` method.

Reply:
@ricardocmj Are you able to provide more detail?

@jspizziri So the race condition mentioned is not applicable?

> This error happens due to a race condition with **hideNotification()**:
> 
> https://github.com/doublesymmetry/KotlinAudio/blob/558fb7f963590813f7e629c73e3c4f9cb29f74f7/kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/notification/NotificationManager.kt#L518-L520
> 
> or with **destroy()**:
> 
> https://github.com/doublesymmetry/KotlinAudio/blob/558fb7f963590813f7e629c73e3c4f9cb29f74f7/kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/notification/NotificationManager.kt#L544-L546

Reply:
Ok sweet :)